src/CTA.WebForms2Blazor/Services/WorkspaceManagerService.cs

```diff
@@ -104,6 +134,17 @@ public DocumentId AddDocument(ProjectId projectId, string documentName, string d
             Document document = _workspace.AddDocument(targetProject.Id, documentName, SourceText.From(documentText));
             _numDocuments += 1;
 
+            if (_numDocuments >= _expectedDocuments)
+            {
+                _allDocumentsInWorkspaceTaskSources.ForEach(source =>
+                {
+                    if (!source.Task.IsCanceled)
+                    {
+                        source.SetResult(true);
+                    }
+                });
+            }
```

Comment:
Same question as the above question but for documents

Reply:
Probably not